gcp/README.md

```diff
@@ -77,3 +77,43 @@ This happens when trying to enable an API that is already enabled. This shouldn'
 ```
 rake xk"[gcloud services disable container.googleapis.com]"
 ```
+
+### Restoring CouchDB data
+
+We are considering number of probable failure scenarios for our GCP infrastructure:
+
+1. **Data corruption on a single CouchDB replica**
+
+In this scenario we rely on CouchDB ability to recover from loss of one or more replicas (our current production CouchDB settings allow us to lose up to 2 random nodes and still keep data integrity). The best course of action as follows:
+
+* Make sure that you figured affected CouchDB pod properly
+* There is a PVC object, associated with affected CouchDB pod. Let's say affected pod is `couchdb-couchdb-1`, then corresponding PVC is `database-storage-couchdb-couchdb-1`, located in the same namespace.
+* Delete associated PVC and then affected pod. For our example case:
+   * `kubectl --namespace gpii delete pvc database-storage-couchdb-couchdb-1`
+   * `kubectl --namespace gpii delete pod couchdb-couchdb-1`
+* After target pod is terminated, Persistent Disk that was mounted into it thru corresponding PVC will be destroyed as well.
+* When new pod is created to replace deleted one, corresponding PVC will be created as well, and, thru it, new PV object for new GCE PD.
+* Please note that you'll have to patch newly created PV, so `k8s-snapshots` can continue to make its snapshots, for our example case:
+   * Get volume name from `kubectl --namespace gpii get pvc database-storage-couchdb-couchdb-1`.
+   * Patch PV with `kubectl patch pv PV_NAME_FROM_PREVIOUS_STEP -p '{"metadata": {"annotations": {"backup.kubernetes.io/deltas": "ENV_DELTAS"}}}'`
+   * To get proper ENV_DELTAS value, check CouchDB Terragrunt module for the environment you're working with (i.e. `live/prd/k8s/gpii/couchdb/terraform.tfvars` for production environment, with deltas `PT5M PT60M PT24H P7D P52W`).
+* CouchDB cluster will replicate data to recreated node automatically.
```

Comment:
I'd add a check to see that the cluster has the nº of the replicas that is expected, and also a check of each replica to see that the cluster is retrieving the same data from each replica. For example, a kubctl command that map a port for each replica, and the curl check should retrieve the same results from each replica.

Reply:
I can only think of something like:
```# rake xk["kubectl exec --namespace gpii -it couchdb-couchdb-0 -c couchdb -- curl -s http://\$TF_VAR_couchdb_admin_username:\$TF_VAR_couchdb_admin_password@127.0.0.1:5984/gpii/"]
docker-compose run --rm --service-ports xk kubectl exec --namespace gpii -it couchdb-couchdb-0 -c couchdb -- curl -s http://$TF_VAR_couchdb_admin_username:$TF_VAR_couchdb_admin_password@127.0.0.1:5984/gpii/
{"db_name":"gpii","update_seq":"209-g1AAAALLeJzLYWBg4MhgTmFwSM4vTc5ISYLRujDaQA9NRC-9IDNTr7gsWS85p7S4JLVILyc_OTEnB2gSUyJDkvz___-zEqWoZ2aSApBMsgcbK4rXWEOSjHUAGRsPNlaWiq5NABlbDzZWmnquzWMBkgwNQApo8vysRDnquRhi9AKI0fsJhTEZRh-AGH2fBq5-ADEaGNaSWQA35vsp","sizes":{"file":268208,"external":150004,"active":118527},"purge_seq":0,"other":{"data_size":150004},"doc_del_count":0,"doc_count":209,"disk_size":268208,"disk_format_version":6,"data_size":118527,"compact_running":false,"cluster":{"q":8,"n":2,"w":2,"r":2},"instance_start_time":"0"}
```
to check DB status on a node. If you have better ideas, please feel free to add them into README and commit into this branch!